File: uds-service/src/main/java/ru/uds/common/controller/ErrorControllerAdvice.java
package ru.uds.common.controller;

import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.ControllerAdvice;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
import ru.uds.common.dto.ErrorResponse;

import javax.persistence.EntityNotFoundException;

@ControllerAdvice
public class ErrorControllerAdvice extends ResponseEntityExceptionHandler {

    @ExceptionHandler(EntityNotFoundException.class)
    public ResponseEntity<ErrorResponse> notFoundException(Exception exception) {
        HttpStatus httpStatus = HttpStatus.NOT_FOUND;
        String message = exception.getMessage() != null ? exception.getMessage() : "Resource not found";

        return new ResponseEntity<>(new ErrorResponse(httpStatus, message), httpStatus);
    }

    @ExceptionHandler(IllegalArgumentException.class)
    public ResponseEntity<ErrorResponse> illegalArgumentException(Exception exception) {
        HttpStatus httpStatus = HttpStatus.BAD_REQUEST;
        String message = exception.getMessage() != null ? exception.getMessage() : "Illegal arguments";

        return new ResponseEntity<>(new ErrorResponse(httpStatus, message), httpStatus);
    }
}

File: tsodd-service/src/main/kotlin/ru/thetenzou/tsoddservice/common/controller/ErrorControllerAdvice.kt
package ru.thetenzou.tsoddservice.common.controller

import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.ControllerAdvice
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler
import ru.thetenzou.tsoddservice.common.dto.ErrorResponse
import javax.persistence.EntityNotFoundException

/**
 * controller advice required for handling exceptions
 */
@ControllerAdvice
class ErrorControllerAdvice : ResponseEntityExceptionHandler() {

    @ExceptionHandler(EntityNotFoundException::class)
    fun notFoundException(exception: Exception) = ResponseEntity(
        ErrorResponse(HttpStatus.NOT_FOUND, exception.message ?: "Resource not found"),
        HttpStatus.NOT_FOUND,
    )

    @ExceptionHandler(IllegalArgumentException::class)
    fun illegalArgumentException(exception: Exception) = ResponseEntity(
        ErrorResponse(HttpStatus.BAD_REQUEST, exception.message ?: "IllegalArguments"),
        HttpStatus.BAD_REQUEST,
    )
}